### PR TITLE
Cassandra improvments

### PR DESCRIPTION
**Pull request by rabbitonweb**
Fixes https://github.com/quasar-analytics/quasar/issues/1676

1. changes the way connectionUri is defined for spark-cassandra
2. adds spark cluster specific configurations for spark-cassandra
2. ~~puts spark-cassandra on travis~~ (some race conditions were detected, will deal with that in separate PR)

### CODE REVIEW

**Comment by rabbitonweb** [comment]
Once this is done we can move spark-cassandra to backend module

**Comment by rabbitonweb** [comment]
@drostron this is ready for review

File: sparkcore/src/main/scala/quasar/physical/sparkcore/fs/cassandra/package.scala [hunk 1]
```diff
@@ -50,52 +51,105 @@ package object cassandra {
...
+            .fold(liftErr("'rootPath' is not a valid path").left[ADir])(_.right[DefinitionError])
+        )
+
+    def fetchParameter(name: String): DefinitionError \/ String = uriOrErr.flatMap(uri =>
+      uri.params.get(name).fold(liftErr(s"'$name' parameter not provided").left[String])(_.right[DefinitionError])
+    )
+
+    for {
+      initSparkConf          <- sparkConfOrErr
+      host                  <- fetchParameter("cassandraHost")
```
Comment (drostron) [inline sparkcore/src/main/scala/quasar/physical/sparkcore/fs/cassandra/package.scala]: Also

File: docker/docker-compose.yml [hunk 2]
```diff
@@ -96,8 +96,9 @@ services:
   
   quasar_spark_cassandra:
     container_name: quasar_spark_cassandra
-    image: cassandra:3.11
+    image: cassandra
```
[the remark this answers is not in the document]
Comment (rabbitonweb) [reply docker/docker-compose.yml]: So we test on latest

File: it/testing.conf.example [hunk 3]
```diff
@@ -8,4 +8,4 @@
 #spark_elastic     ="spark://<spark_host>:<spark_port>?rootPath=<root_path>"
 #marklogic         ="xcc://<username>:<password>@<host>:<port>/<database>"
 #couchbase         ="couchbase://<host>[:<port>]/<bucket-name>?password=<password>&docTypeKey=<type>[&queryTimeoutSeconds=<seconds>]"
-#spark_cassandra   ="spark://<spark_host>:<spark_port>|<cassandra_host>|<root_path>"
+#spark_cassandra   ="spark://<sparkUri>?cassandraHost=<host>t&cassandraPort=<port>&rootPath=<rootPath>"
```
[the remark this answers is not in the document]
Comment (rabbitonweb) [reply it/testing.conf.example]: It is exactly the same thing. I can unify it so there is no confusion

File: docker/docker-compose.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (drostron) [reply docker/docker-compose.yml]: Could we just update to the latest version for stability and consistency with the other docker compose entries?

[the remark this answers is not in the document]
Comment (rabbitonweb) [reply docker/docker-compose.yml]: 👍

**Comment by rabbitonweb** [comment]
@drostron changes are made

**Comment by drostron** [comment]
Looks good. Needs a merge from master.

File: it/src/main/resources/tests/mongodb/temporal/filterOnDate.test [hunk 4]
```diff
@@ -0,0 +1,18 @@
+{
```
Comment (jsantos17) [inline it/src/main/resources/tests/mongodb/temporal/filterOnDate.test]: btw I think this file is duplicating the one that alreday exists [here](https://github.com/rabbitonweb/quasar/blob/cassandra-improvments/it/src/main/resources/tests/temporal/filterOnDate.test)

Comment (rabbitonweb) [reply it/src/main/resources/tests/mongodb/temporal/filterOnDate.test]: there few other cases like that - i think there is some distinction for some reason

**Comment by rabbitonweb** [comment]
@jsantos17 i think those tests were different after all, CI failed but it is 1am and I'm going to bed

**Pull request merged**